pkg/volume/host_path/host_path.go

```diff
@@ -132,6 +121,9 @@ func (plugin *hostPathPlugin) NewDeleter(spec *volume.Spec) (volume.Deleter, err
 }
 
 func (plugin *hostPathPlugin) NewProvisioner(options volume.VolumeOptions) (volume.Provisioner, error) {
+	if !plugin.config.ProvisioningEnabled {
```

Comment:
There was a use-case for it in V1 (testing and.. i forget the other) so we left it.

There's an interesting use case in issue #27732 asking to use node level storage as a local cache.